SYCL/ESIMD/esimd_check_vc_codegen.cpp

```diff
@@ -0,0 +1,104 @@
+//==-------- esimd_check_vc_codegen.cpp - DPC++ ESIMD on-device test -------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+// REQUIRES: gpu
+// UNSUPPORTED: cuda || hip
+// esimd_emulator does not support online-compiler that invokes 'piProgramBuild'
+// UNSUPPORTED: esimd_emulator
+// RUN: %clangxx -fsycl %s -o %t.out
+// RUN: env SYCL_PI_TRACE=-1 %GPU_RUN_PLACEHOLDER %t.out 2>&1 %GPU_CHECK_PLACEHOLDER
+
+#include "esimd_test_utils.hpp"
+
+#include <CL/sycl.hpp>
+#include <iostream>
+#include <sycl/ext/intel/experimental/esimd.hpp>
+
+using namespace cl::sycl;
+
+int main(void) {
+  constexpr unsigned Size = 1024 * 128;
+  constexpr unsigned VL = 16;
+
+  float *A = new float[Size];
+  float *B = new float[Size];
+  float *C = new float[Size];
+
+  for (unsigned i = 0; i < Size; ++i) {
+    A[i] = B[i] = i;
+    C[i] = 0.0f;
+  }
+
+  try {
```

Comment:
wouldn't an empty kernel run via `single_task` (or `parallel_for` if single_task is not supported yet) suffice here?

Reply:
I'm not sure if it is right to modify this test as it is duplicate of `vadd_1d.cpp`. Original `vadd_1d.cpp` is changed to remove `CHECK` commands for `piProgramBuild` and to mark `XFAIL: esimd_emulator`. This test has those `CHECK` commands and `UNSUPPORTED: esimd_emulator` as `esimd_emulator` does not support online compilation

Do you want this test to be removed like `sycl_esimd_mix_check_build_opts.cpp` and original `vadd_1d.cpp` to have `UNSUPPORTED: esimd_emulator`?

Reply:
see suggested changes

Reply:
Suggestions applied.